src/link/index.ts

```diff
@@ -166,6 +168,61 @@ export default async function linkPackages (
   }
 }
 
+async function shamefullyFlattenTree(
+  flatResolvedDeps: DependencyTreeNode[],
+  opts: {
+    force: boolean,
+    dryRun: boolean,
+    baseNodeModules: string,
+    bin: string,
+    pkg: PackageJson,
+    outdatedPkgs: {[pkgId: string]: string},
+    wantedShrinkwrap: Shrinkwrap,
+  },
+) {
+  const pkgNamesExcludedFromFlattening = {}
+  // first of all, exclude the root packages, as they are already linked
```

Comment:
this part might be an issue because the key in specifiers can be an alias. Better just to use the second part. You can easily exclude packages that have depth 0

Reply:
it is actually what I want, I want to use the alias because I don't want to override the symlink, which is named with the _alias_, not with the real name. I can't use the previous code because it only contains newly installed packages, the test I just wrote would fail.